test/test_transforms.py

```diff
@@ -1403,6 +1322,88 @@ def test_random_erasing(self):
         t.__repr__()
 
 
+@pytest.mark.parametrize('channels', [1, 3, 4])
+def test_pil_to_tensor(channels):
+    height, width = 4, 4
+    trans = transforms.PILToTensor()
+
+    with pytest.raises(TypeError):
+        trans(np.random.rand(1, height, width).tolist())
+        trans(np.random.rand(1, height, width))
```

Comment:
I think we should have 2 `raises` blocks here, one per line. This is because we want to make sure that both lines below raise a TypeError. With a  single block, we're only asserting that *one* of them raises the error. In fact I think the original test code was incorrect.

Also, we don't need those assertions to be part of the parametrization, it's a bit overkill as they don't need the `channel` parameter. So I'd suggest to  extract them into a separate `test_pil_to_tensor_errors()` test function